packages/ember-data/lib/system/snapshot.js

```diff
@@ -355,6 +355,14 @@ Snapshot.prototype = {
   */
   unknownProperty: function(keyName) {
     return this.get(keyName);
+  },
+
+  /**
+    @method _createSnapshot
+    @private
+  */
+  _createSnapshot: function() {
+    Ember.assert("You called _createSnapshot on what's already a DS.Snapshot. You shouldn't manually create snapshots in your adapter since the store passes snapshots to adapters by default.");
```

Comment:
@wecc Sorry for changing my mind here. I chatted with Igor about this and it might be a nicer migration path for this method to give a deprecation warning and just return the instance of the snapshot itself.